dataset/src/main/scala/frameless/TypedDatasetForwarded.scala

```diff
@@ -65,6 +92,66 @@ trait TypedDatasetForwarded[T] { self: TypedDataset[T] =>
   def repartition(numPartitions: Int): TypedDataset[T] =
     TypedDataset.create(dataset.repartition(numPartitions))
 
+
+  /**
+    * Get the [[TypedDataset]]'s current storage level, or StorageLevel.NONE if not persisted.
+    *
+    * apache/spark
+    */
+  def storageLevel(): StorageLevel =
+    dataset.storageLevel
+
+  /**
+    * Returns the content of the [[TypedDataset]] as a Dataset of JSON strings.
+    *
+    * apache/spark
+    */
+  def toJSON: TypedDataset[String] =
+    TypedDataset.create(dataset.toJSON)
+
+  /**
+    * :: Experimental ::
+    * Defines an event time watermark for this [[TypedDataset]]. A watermark tracks a point in time
+    * before which we assume no more late data is going to arrive.
+    *
+    * Spark will use this watermark for several purposes:
+    *  - To know when a given time window aggregation can be finalized and thus can be emitted when
+    *    using output modes that do not allow updates.
+    *  - To minimize the amount of state that we need to keep for on-going aggregations,
+    *    `mapGroupsWithState` and `dropDuplicates` operators.
+    *
+    *  The current watermark is computed by looking at the `MAX(eventTime)` seen across
+    *  all of the partitions in the query minus a user specified `delayThreshold`.  Due to the cost
+    *  of coordinating this value across partitions, the actual watermark used is only guaranteed
+    *  to be at least `delayThreshold` behind the actual event time.  In some cases we may still
+    *  process records that arrive more than `delayThreshold` late.
+    *
+    * @param eventTime the name of the column that contains the event time of the row.
+    * @param delayThreshold the minimum delay to wait to data to arrive late, relative to the latest
+    *                       record that has been processed in the form of an interval
+    *                       (e.g. "1 minute" or "5 hours"). NOTE: This should not be negative.
+    *
+    * apache/spark
+    */
+  def withWatermark(eventTime: String, delayThreshold: String): TypedDataset[T] =
```

Comment:
This looks a bit scary, I think we shouldn't include it without first understanding it. Given than it's currently untested I propose to postpone it to another PR.

I gave [this](https://databricks.com/blog/2017/05/08/event-time-aggregation-watermarking-apache-sparks-structured-streaming.html) a quick read and looks like this method adds an extra column to the dataset, probably something that we want to model in the types one way or another...

Reply:
I would also leave `writeStream` out for now, it looks like it belong to some some part of Spark that we didn't touch at all yet.

Reply:
[More doc](https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html) an Spark streaming. We should maybe create a dedicated issue for these methods and remove them from #163